### PR TITLE
Cleanup test data also on failure

### PR DESCRIPTION
**Pull request by esoergel**
https://github.com/dimagi/commcare-hq/pull/30309/files#r704240324

Tests only

### CODE REVIEW

File: corehq/apps/es/tests/test_case_search_es.py [hunk 1]
```diff
@@ -318,6 +318,11 @@ def _assert_query_runs_correctly(self, domain, input_cases, query, xpath_query,
                 output
             )
 
+    def _create_case_search_config(self):
+        config, _ = CaseSearchConfig.objects.get_or_create(pk=self.domain, enabled=True)
+        self.addCleanup(config.delete)
```
Comment (esoergel) [inline corehq/apps/es/tests/test_case_search_es.py:323]: is this more clever than helpful?

Comment (snopoke) [reply corehq/apps/es/tests/test_case_search_es.py:323]: seems ok

**Review by snopoke — APPROVED** [review]

**Comment by millerdev** [comment]
I don't think it's necessary to explicitly clean up SQL data (except in shard dbs) that is created by `TestCase` subclasses. Each test is run in a transaction which is then rolled back after the test completes. Is there some special circumstance for these tests that requires explicit cleanup?

**Comment by snopoke** [comment]
> I don't think it's necessary to explicitly clean up SQL data (except in shard dbs) that is created by `TestCase` subclasses. Each test is run in a transaction which is then rolled back after the test completes. Is there some special circumstance for these tests that requires explicit cleanup?

That is true, I guess it's just a habbit. But also better to use this method than call 'delete' after the assert.

**Pull request merged**